osu.Game/Screens/Play/PlayerLoader.cs

```diff
@@ -228,7 +228,7 @@ private void load(SessionStatics sessionStatics, OsuConfigManager config)
                 sampleRestart = new SkinnableSound(new SampleInfo(@"Gameplay/restart", @"pause-retry-click"))
             };
 
-            if (Beatmap.Value.BeatmapInfo.EpilepsyWarning)
+            if (Beatmap.Value.Beatmap.EpilepsyWarning)
```

Comment:
Have you tested whether this is blocking on the beatmap load? This looks kinda scary to me.

Reply:
I haven't tested this until now (see mention in OP that this probably needs more testing). I was interested in getting the shape of this right first and foremost, and was only going to go over specifics later. The extent of what is to be moved out is negotiable.

As far as the answer goes, I believe it to be "generally it will not be", because the game-wide async load process will have already completed by now:

https://github.com/ppy/osu/blob/57688c2cfadc1b8322527016e8a4626176074904/osu.Game/OsuGame.cs#L799-L803

But sure, I can agree with not moving this out. I was in two minds about doing this myself, but am prepared to revert this part. It just seemed to me like it wasn't impossible to move it so I did it, to cover absolutely everything that can be moved without significant changes.

Reply:
> we may want this to be databased as metadata to show at song select or be searchable. In such a case, we wouldn't want to move it out of `BeatmapInfo`, correct?

That is my understanding yes.